### DISCUSSION TITLE
[Feature] Auto pause jobs based on shell command (e.g. temperature too high)

### DISCUSSION THREAD

**Discussion by Explosion-Scratch** [immich-app/immich#7149] [Feature Request]
### The feature

It would be great if you could configure an auto pause command, so for example if the command returns anything then immich pauses jobs. This command could be run every 30 seconds or so.

Examples:

Detect if any sensor temp is > 65deg C:
```
count=$(cat /sys/class/thermal/thermal_zone*/temp | awk '$1 > 65000' | wc -l); if [ $count -gt 0 ]; then echo "Temperature is above 65"; else echo ""; fi
```

Test if CPU usage > 150%:
```
top -bn1 | awk 'NR>7{s+=$9} END {if (s > 150) print "CPU usage is above 150%"; else print ""}'
```

**Comment by bo0tzz** [comment]
Since Immich is running in a container, it probably doesn't have access to sensors and the like. That said, this is something that can be easily slapped together into a bash script that calls the Immich [API to pause the jobs](https://immich.app/docs/api/send-job-command), which is way more flexible than building it into Immich.

**Comment by alextran1502** [comment]
There are APIs to controlling the JOB already, you can make curl requests to it https://immich.app/docs/api/send-job-command

**Comment by Explosion-Scratch** [comment]
Thanks so much, didn't know about that API! I'll make a Bash script and post it here in case somebody comes across this via google sometime

**Comment by Explosion-Scratch** [comment]
In case anybody wanted it, here's a quick Node.js Script I made:

https://gist.github.com/Explosion-Scratch/56fd5374e3e66999ffc0621874906274

This pauses immich jobs on several relevant conditions
